doc/specs/#16599 - Quick Fix.md

```diff
@@ -0,0 +1,43 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2024-04-01
+last updated: 2024-04-02
+issue id: "#16599"
+---
+# Quick Fix
+## Solution Design
+Quick Fix will be a new UI surface that allows the user to interact with the terminal and leverage the context of a specific command being executed. The UI is discussed in the [UI/UX Design section](#ui/ux-design).
+
+The UI itself will live in the TerminalApp project, whereas any logic populating the UI will live in the TerminalControl project. This allows for any consumer of the TerminalControl project to still have access to these features, but have the ability to represent them in whatever UI they please. To accomplish this, the TerminalControl layer must expose a `IVector<String> GetQuickFixContext(UInt32 Row)` which will retrieve all of the quick fix suggestions at the given `Row`. The returned suggestions will be consumed by the TerminalApp layer, where the `IVector<String>` will be converted to "sendInput" actions and displayed in the quick fix UI.
+
+If and when the buffer resizes and reflows, the TerminalApp layer will be notified that the "contexts" are now invalidated. At that point, TerminalApp will call `GetQuickFixContext()` again to retrieve the updated information.
+
+The actual suggestions will be populated from various sources such as the following:
+- WinGet Command Not Found OSC Sequence:
+	- `OSC 9001; CmdNotFound; <missingCmd>` will be used by the attached CLI app to notify the terminal that a command `missingCmd` wasn't found. The TermControl layer will use WinGet's API to search for any packages that would remediate this error and construct a list of suggested packages to be installed.
+`OSC 9001` can be expanded on to add more suggestions via VT sequences from the underlying CLI app; for more on this, read the [Future considerations section](#future-considerations).
+## UI/UX Design
+A Quick Fix icon will be displayed immediately left of the prompt. If the icon fits in the left-gutter region, it will be drawn there. Otherwise, a cropped and transparent version will be displayed overlapping the text region; when the icon is hovered over, the icon will appear more clear by becoming more opaque and enlarging to become uncropped.
```

Comment:
nit: what is the "gutter"?

_I'm nitting here because I know we discussed this in team sync, but we should probably translate that discussion into the text here_

Reply:
> cropped and transparent version

that makes it sound like you'd be doing:
![b4cfa47c-0330-4338-94dc-84f955b55af3](https://github.com/microsoft/terminal/assets/18356694/8db5e428-d8e5-41c4-b104-6232583e7295)

and not
![image](https://github.com/microsoft/terminal/assets/18356694/8231c223-e45c-4e9d-8f0c-b993391ab5a3)